native/iparmq.go

```diff
@@ -0,0 +1,112 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "math"
+
+// Iparmq can return problem and machine dependent parameters useful for Dhseqr
+// and related subroutines for eigenvalue problems. It must hold for ispec that
+//  12 <= ispec <= 16
+// otherwise Iparmq will panic.
+//
+// ispec specifies the parameter to return:
+//  12: Crossover point between Dlahqr and Dlaqr0. Will be at least 11.
+//  13: Deflation window size.
+//  14: Nibble crossover point.
+//  15: Number of simultaneous shifts in a multishift QR iteration.
+//  16: Select structured matrix multiply.
+// For other values of ispec Iparmq will panic.
+//
+// name is the name of the calling function. name must be in uppercase but this
+// is not checked.
+//
+// n is the order of the Hessenberg matrix H.
+//
+// ilo and ihi specify the block [ilo:ihi+1,ilo:ihi+1] that is being processed.
+//
+// lwork is the amount of workspace available.
+//
+// Except for ispec input parameters are not checked.
+//
+// Iparmq is an internal routine. It is exported for testing purposes.
+func (Implementation) Iparmq(ispec int, name, _ string, n, ilo, ihi, lwork int) int {
```

Comment:
`opts` is described as `concatenation of the string arguments to TTQRE`. From `[1]`, TTQRE was an experimental Fortran implementation of the multishift QR algorithm by Braman et al. and `opts` is not used in the reference LAPACK's IPARMQ. This suggests that it could be removed. However, DLAQR4 does assemble a string based on `wantt` and `wantz` so I can imagine that `opts` could be used to tune the returned values somehow (although I have no idea how). With this, let me know if I should remove it or keep it.